### PR TITLE
Use main thread pool node-runner as CPU-bound worker

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
This is a follow-up PR for https://github.com/rchain/rchain/pull/1595
The previous PR didn't pass the integration tests on Gitlab. This PR takes a further step in optimizing the thread pools.

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
https://rchain.atlassian.net/browse/CORE-1230

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] If this PR adds a new feature, this PR includes tests related to this feature.
- [x] You assigned one person to review this PR

### CODE REVIEW

File: comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala [hunk 1]
```diff
@@ -211,7 +211,7 @@ class TcpTransportLayer(host: String, port: Int, cert: String, key: String, maxM
     Task.delay {
       new TcpServerObservable(port, serverSslContext, maxMessageSize)
         .mapParallelUnordered(parallelism)(dispatchInternal)
-        .subscribe()(Scheduler.computation(parallelism, "tl-dispatcher"))
+        .subscribe()(Scheduler.fixedPool("tl-dispatcher", parallelism))
```
Comment (goral09) [inline comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala]: Why did you choose `fixedPool` ? Isn't the TCP layer non-blocking? If so then why you didn't choose `Scheduler.io` ? Why not _cached_ thread pool?

File: node/src/main/scala/coop/rchain/node/Main.scala [hunk 2]
```diff
@@ -27,14 +27,18 @@ object Main {
 
   def main(args: Array[String]): Unit = {
 
+    implicit val scheduler: Scheduler = Scheduler.computation(
```
Comment (goral09) [inline node/src/main/scala/coop/rchain/node/Main.scala]: Why `computation`? It uses `ForkJoinPool` internally, is `mainProgram` of this _fork-join_ characteristic?

File: node/src/main/scala/coop/rchain/node/api/grpc.scala [hunk 3]
```diff
@@ -35,23 +40,28 @@ object GrpcServer {
     Task.delay {
       NettyServerBuilder
         .forPort(port)
-        .executor(scheduler)
+        .executor(grpcExecutor)
```
Comment (goral09) [inline node/src/main/scala/coop/rchain/node/api/grpc.scala]: What is this `executor` used for? You set manually the `grpcExecutor` betlow for `ReplGrpcMonix` and `DiagnosticsGprcMonix` manually so for which calls is, this `executor` call, setting the scheduler?

File: node/src/main/scala/coop/rchain/node/configuration/commandline/Options.scala [hunk 4]
```diff
@@ -251,7 +251,7 @@ final case class Options(arguments: Seq[String]) extends ScallopConf(arguments)
       opt[Int](descr = "Maximum size of message that can be sent via transport layer")
 
     val threadPoolSize =
-      opt[Int](descr = "Maximum number of threads used by rnode")
+      opt[Int](descr = "Maximum number of threads used by rnode", hidden = true)
```
Comment (goral09) [inline node/src/main/scala/coop/rchain/node/configuration/commandline/Options.scala]: If `hidden = true` hides this flag from the help output why do we need this option in the first place?

File: comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala (hunk 1 above)
Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala]: First of all, this is the thread pool for consuming the message queue and dispatching the messages to Casper and Connect. There shouldn't be any IO here and I expect the work to be CPU-bound. On the other hand, a `Scheduler.compution` thread pool is backed by a `ForkJoinPool` which potentially creates and removes threads. We need a fixed size thread pool with `Math.max(Runtime.getRuntime.availableProcessors(), 2)` thread. Maybe we can at some point in future use the global `node-runner` thread pool.

File: node/src/main/scala/coop/rchain/node/api/grpc.scala (hunk 3 above)
Comment (sebtomba) [reply node/src/main/scala/coop/rchain/node/api/grpc.scala]: This is `grpc-java` specific. The executor is used internally for incoming requests. If we don't set here an `executor` then `grpc-java` will create its own `ForkJoinPool`. We want to use the same thread pool for all gRPC IO in `ReplGrpcMonix`, `DiagnosticsGprcMonix`, `DeployGrpcService` and `TcpTransportLayer`.

File: node/src/main/scala/coop/rchain/node/configuration/commandline/Options.scala (hunk 4 above)
Comment (sebtomba) [reply node/src/main/scala/coop/rchain/node/configuration/commandline/Options.scala]: I want to remove this flag in the future. But ATM I'm just hiding it to not break some existing scripts. This flag is like _depricated_.

**Pull request merged**